lib/cfg-tree.c

```diff
@@ -645,6 +646,59 @@ log_expr_node_lookup_flag(const gchar *flag)
   return 0;
 }
 
+static void
+_log_pipe_unref(void *s)
+{
+  log_pipe_unref(s);
+}
+
+static void
+cfg_tree_start_registering_pipes_with_persist_name(CfgTree *self)
+{
+  main_loop_assert_main_thread();
+
+  self->pipes_with_persis_name = g_hash_table_new_full(NULL, NULL, NULL, _log_pipe_unref);
+}
+
+static void
+cfg_tree_stop_registering_pipes_with_persist_name(CfgTree *self)
+{
+  if (self->pipes_with_persis_name)
+    {
+      main_loop_assert_main_thread();
+
+      g_hash_table_unref(self->pipes_with_persis_name);
```

Comment:
Iterate over and unref the logpipes inside just in case something forgots to do it before this call

Reply:
it is automatic if you use g_hash_table_new_full

https://docs.gtk.org/glib/type_func.HashTable.unref.html
https://docs.gtk.org/glib/type_func.HashTable.new_full.html